File: atcoder/abc168/D/main.py
#!/usr/bin/env python3
import sys
import collections

sys.setrecursionlimit(10 ** 8)
ni = lambda: int(sys.stdin.readline())
nm = lambda: map(int, sys.stdin.readline().split())
nl = lambda: list(nm())
ns = lambda: sys.stdin.readline().rstrip()

N, M = nm()
adj = [[] for _ in range(N)]
for i in range(M):
    a, b = nm()
    a -= 1
    b -= 1
    adj[a].append(b)
    adj[b].append(a)


def solve():
    INF = 1 << 62
    q = collections.deque([(0, 0, -1)])
    dist = [INF] * N
    prev = [None] * N
    while q:
        l, node, pnode = q.popleft()
        if dist[node] < l:
            continue
        dist[node] = l
        prev[node] = pnode
        for nxt in adj[node]:
            if nxt != pnode and dist[nxt] > l + 1:
                q.append((l + 1, nxt, node))
    # if any(x is None for x in prev):
    #     print("No")
    #     return
    print("Yes")
    for i in range(1, N):
        print(prev[i] + 1)


if __name__ == "__main__":
    solve()

File: atcoder/abc168/D/main.cpp
#pragma GCC optimize("O3")
#include <bits/stdc++.h>
#define REP(i, n) for (decltype(n) i = 0; i < (n); ++i)
using namespace std;
using i64 = long long;
using u64 = unsigned long long;
int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);
  int N, M;
  cin >> N >> M;
  vector<vector<int>> adj(N);
  REP(i, N) { adj[i].reserve(M); }
  REP(i, M) {
    int a, b;
    cin >> a >> b;
    a--;
    b--;
    adj[a].push_back(b);
    adj[b].push_back(a);
  }
  deque<tuple<int, int, int>> q;
  q.emplace_back(0, 0, -1);
  int INF = N + 10;
  vector<int> dist(N, INF);
  vector<int> prev(N, -1);
  int cnt = 0;
  while (!q.empty()) {
    auto [l, node, pnode] = q.front();
    q.pop_front();
    if (dist[node] <= l) continue;
    dist[node] = l;
    prev[node] = pnode;
    ++l;
    for (auto&& nxt : adj[node]) {
      if (dist[nxt] > l) {
        q.emplace_back(l, nxt, node);
      }
    }
  }
  cout << "Yes\n";
  for (int i = 1; i < N; ++i) {
    cout << (prev[i] + 1) << "\n";
  }
}
